@PROTOBUF_SYNTAX_VERSION@
import "dccl/protobuf/option_extensions.proto";

message MiniUser
{
  option (dccl.msg).id = 1000001;
  option (dccl.msg).max_bytes = 2;
  
  required uint32 user = 1 [(dccl.field).min=0,
                            (dccl.field).max=0x03FF,
                            (dccl.field).in_head=true];
}

message MiniOWTT
{
  option (dccl.msg).id = 1000002;
  option (dccl.msg).max_bytes = 2;
  
  required uint32 clock_mode = 1 [(dccl.field).min=0,
                                  (dccl.field).max=3,
                                  (dccl.field).in_head=true];
  
  required uint32 tod = 2 [(dccl.field).min=0,
                           (dccl.field).max=0x0F,
                           (dccl.field).in_head=true];
  
  required uint32 user = 3 [(dccl.field).min=0,
                            (dccl.field).max=0x0F,
                            (dccl.field).in_head=true];
}

message MiniAbort
{
  option (dccl.msg).id = 1000003;
  option (dccl.msg).max_bytes = 2;
  
  required uint32 user = 1 [(dccl.field).min=0,
                            (dccl.field).max=0x03FF,
                            (dccl.field).in_head=true];
}

message NormalDCCL
{
  option (dccl.msg).id = 1;
  option (dccl.msg).max_bytes = 32;

  required int32 a = 1 [(dccl.field).min=0,
                        (dccl.field).max=0xFFFF];
  required int32 b = 2 [(dccl.field).min=0,
                        (dccl.field).max=0xFFFF];
  
}

